### PR TITLE
ADR 3: kURL External Add-On Capability

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What this PR does / why we need it:

Document the architectural design decision retroactively.

### CODE REVIEW

**Review by laverya — DISMISSED** [review]

File: docs/arch/adr-003-external-addons.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    s3Override: https://s3.kurl.sh/external/kotsadm-1.85.0-a482541.tar.gz
+```
+
+## Status
+
+Proposed
+
+## Consequences
+
+Once an version is published to an external versions.json file, kURL will copy that add-on package version to its object storage bucket.
```
Comment (ricardomaraschini) [inline docs/arch/adr-003-external-addons.md:51]: How kurl will know what are the external add-ons and respective URLs for retrieving the version.json file ?

**Review by ricardomaraschini — DISMISSED** [review]

File: docs/arch/adr-003-external-addons.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+```yaml
+spec:
+  kotsadm:
+    version: 1.85.0
+    s3Override: https://s3.kurl.sh/external/kotsadm-1.85.0-a482541.tar.gz
+```
+
+## Status
+
+Proposed
```
Comment (ricardomaraschini) [inline docs/arch/adr-003-external-addons.md:47]: May be a weird question but: it is "Proposed" while it is not merged but once it is merged it is "Accepted" ?

File: docs/arch/adr-003-external-addons.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+When a kURL spec is resolved as part of the script rendering process, external add-on package versions will be rendered including the `s3Override` property with reference to the package.
+kURL will use this property to identify the source and download the add-on package when installing the add-on in the end user's environment.
+
+```yaml
+spec:
+  kotsadm:
+    version: 1.85.0
+    s3Override: https://s3.kurl.sh/external/kotsadm-1.85.0-a482541.tar.gz
+```
```
Comment (rrpolanco) [inline docs/arch/adr-003-external-addons.md:43]: Can we move these details to the **Consequences** section? The **Decision** subheading should be short and concise.

File: docs/arch/adr-003-external-addons.md (hunk 2 above)
Comment (rrpolanco) [reply docs/arch/adr-003-external-addons.md:47]: Once the team approves this PR and comments have been resolved, then a subsequent update should be made to update the **Status** to _Accepted_ and then merging can proceed.

File: docs/arch/adr-003-external-addons.md (hunk 3 above)
Comment (rrpolanco) [reply docs/arch/adr-003-external-addons.md:43]: Yes, there's no hard and fast rule for additional context in the decision section but in my experience and in practice, this section should only state the final decision. This makes the _Decision_ subheading cleaner and concise for the purpose of better readability when skimming through an ADR repository (and enables ADR tooling to just pull decision verbiage only). Moreover, the why&how details seem to fit better in the first subheading (general context) or perhaps in a new subheading which is all useful information which leads to a decision.

Comment (emosbaugh) [reply docs/arch/adr-003-external-addons.md:43]: i prefer to keep context as the "why". I feel like the current template there is no place for "how". I like the idea of a new subheading. i will look at other adr templates and make a proposal.

Comment (emosbaugh) [reply docs/arch/adr-003-external-addons.md]: @rrpolanco proposed changes included in this commit

https://github.com/replicatedhq/kURL/pull/3543/commits/1728e3d31dd5ef653899fe1cb9fec8cd19bc027c

File: docs/arch/README.md [hunk 4]
```diff
@@ -44,6 +44,14 @@ section is value-neutral. It is simply describing facts.
 This section describes our response to these forces. It is stated in full sentences,
 with active voice. "We will ..."
 
+## Solution
+
+Optional.
+
+More details explaining our response.
```
Comment (rrpolanco) [inline docs/arch/README.md:51]:
```suggestion
Further details explaining our decision.
```

File: docs/arch/adr-003-external-addons.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+## Status
+
+Proposed
+
+## Consequences
+
+Once an version is published to an external versions.json file, kURL will copy that add-on package version to its object storage bucket.
```
Comment (rrpolanco) [inline docs/arch/adr-003-external-addons.md:55]:
```suggestion
Once a version is published to an external versions.json file, kURL will copy that add-on package version to its object storage bucket.
```

**Review by rrpolanco — APPROVED** [review]
Looks good. The new headings in the ADR template are helpful, thanks for adding those.

**Comment by rrpolanco** [comment]
@emosbaugh You forgot to update the status to `Accepted`.

**Pull request merged**